Formula/glib-utils.rb

```diff
@@ -0,0 +1,112 @@
+class GlibUtils < Formula
+  include Language::Python::Shebang
+
+  desc "Python utilities for GLib"
+  homepage "https://developer.gnome.org/glib/"
+  url "https://download.gnome.org/sources/glib/2.72/glib-2.72.2.tar.xz"
+  sha256 "78d599a133dba7fe2036dfa8db8fb6131ab9642783fc9578b07a20995252d2de"
+  license "LGPL-2.1-or-later"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on "python@3.10"
+
+  def install
+    # Point the headers and libraries to `glib`.
+    # The headers and libraries will be removed later because they are provided by `glib`.
+    glib = Formula["glib"]
+    args = std_meson_args.delete_if do |arg|
+      arg.start_with?("--prefix=", "--includedir=", "--libdir=")
+    end
+    args += %W[
+      --prefix=#{prefix}
+      --includedir=#{glib.opt_include}
+      --libdir=#{glib.opt_lib}
+    ]
+
+    # Disable dtrace; see https://trac.macports.org/ticket/30413
+    # and https://gitlab.gnome.org/GNOME/glib/-/issues/653
+    args += %W[
+      --default-library=both
+      --localstatedir=#{var}
+      -Diconv=auto
+      -Dgio_module_dir=lib/gio/modules
+      -Dbsymbolic_functions=false
+      -Ddtrace=false
+    ]
+
+    mkdir "build" do
+      system "meson", *args, ".."
+      system "ninja", "-v"
+
+      # Skip files already in glib
+      Formula["meson"].opt_libexec.cd do
+        system "bin/python3", "-c", pyscript
+      end
+      system "ninja", "install", "-v"
```

Comment:
This approach looks like we're going to end up building glib twice across two different formulae. Why aren't we just following Gentoo's approach here?

Reply:
> Why aren't we just following Gentoo's approach here? 

Commented in https://github.com/Homebrew/homebrew-core/pull/103366#issuecomment-1157310697 for potential future updates. The Gentoo's approach uses fixed paths.

Reply:
Do we need to build glib twice to check for Python scripts?

Reply:
> Do we need to build glib twice to check for Python scripts?

Yes, I think.

1. There are several gdb integration scripts depend on the `libdir` which points to the prefix of `glib` rather than `glib-utils`.
2. We don't know the future release would look like. Which files will be installed into the prefix? Will the scripts contain hardcoded paths depend on the configure options?
3. Maybe we can read the meson's install plan directly to determine which Python scripts will be installed. However, meson also supports install a entire subdir rather than s individual files separately. We may miss something in the future release.

Reply:
I'm fine with building `glib` twice. As this happens only on our CI, and the build time is around 15 min in total, this should be transparent for most of our users. People compiling from source in non-standard prefixes are already used to wait for stuff to compile I guess.

I must say the whole approach looks a little bit heavy, but I think it's the safest way to implement this change. I'm fine with the actual approach as it is right now.